### PR TITLE
Improve editor speed calculation

### PR DESCRIPTION
**Pull request by Holomanga**
In the editor, the speed calculation behaves oddly ([posted about this in the forum](https://community.revolutionarygamesstudio.com/t/0-5-0-public-beta-godot-version/2617/35?u=holomanga)).

I've altered it to calculate the total force (from membrane + organelles with movement component) and then divide by the microbe mass.

### CODE REVIEW

**Comment by Holomanga** [comment]
The actual force applied by a flagella seems to be about 15, precisely 20% of the value of FLAGELLA_BASE_FORCE (75).

**Comment by hhyyrylainen** [comment]
I think what might be happening with the flagella (and what you should probably include in the calculation) is that the force of the flagella depend on how parallel they are to the movement direction. So a flagellum pointing forward contributes no force to trying to move forward.

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 1]
```diff
@@ -636,26 +636,27 @@ public void RemoveOrganelle()
     public float CalculateSpeed()
     {
         float finalSpeed = 0;
-        int flagCount = 0;
-        float lengthMicrobe = 0;
+        float massMicrobe = Constants.MICROBE_BASE_MASS;
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:639]: `microbeMass` sound better to me.

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 2]
```diff
@@ -636,26 +636,27 @@ public void RemoveOrganelle()
     public float CalculateSpeed()
     {
         float finalSpeed = 0;
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:638]: this variable should be declared only once needed.

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 3]
```diff
@@ -636,26 +636,27 @@ public void RemoveOrganelle()
...
         foreach (var organelle in editedMicrobeOrganelles.Organelles)
         {
-            lengthMicrobe += organelle.Definition.HexCount;
+            massMicrobe += organelle.Definition.Mass;
 
-            // TODO: this should be changed to instead check for the
-            // movement component presence
-            if (organelle.Definition.InternalName == "flagellum")
-                flagCount++;
+            if (organelle.Definition.Components.Movement != null)
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:648]: This check should be:
`organelle.Definition.HasComponentFactory<MovementComponentFactory>()`

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 4]
```diff
@@ -635,27 +635,38 @@ public void RemoveOrganelle()
...
+                Vector3 organelleDirection = (Hex.AxialToCartesian(new Hex(0, 0))
+                 - Hex.AxialToCartesian(organelle.Position)).Normalized();
+
+                float directionFactor = organelleDirection.Dot(forwardsDirection);
 
-            // TODO: this should be changed to instead check for the
-            // movement component presence
-            if (organelle.Definition.InternalName == "flagellum")
-                flagCount++;
+                // Flagellae pointing backwards don't slow you down
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:656]: Plural of flagellum is `Flagella`

**Pull request merged**